scala/lambdas/ingest-flow-control/src/main/scala/uk/gov/nationalarchives/ingestflowcontrol/Lambda.scala

```diff
@@ -100,13 +132,20 @@ class Lambda extends LambdaRunner[Input, Unit, Config, Dependencies] {
                       .void
                 case None =>
                   val remainingSystems = sourceSystems.filter(_.systemName != systemToStartTaskOn)
+                  logger.info(
+                    Map("executionName" -> s"""${potentialInput.map(_.executionName)}""", "remainingSystems" -> s"""${remainingSystems.map(_.systemName).mkString(", ")}""")(
+                      s"System $systemToStartTaskOn does not have a waiting task"
+                    )
+                  )
                   startTaskBasedOnProbability(remainingSystems)
             }
     }
 
-    def writeItemToQueueTable(flowControlConfig: FlowControlConfig): IO[Unit] = IO.whenA(input.taskToken.nonEmpty) {
+    def writeItemToQueueTable(flowControlConfig: FlowControlConfig): IO[Unit] = IO.whenA(potentialInput.nonEmpty) {
+      val input = potentialInput.get
       val sourceSystemName = input.executionName.split("_").head
       val systemName = flowControlConfig.sourceSystems.find(_.systemName == sourceSystemName).map(_.systemName).getOrElse(default)
+      logger.info(Map("executionName" -> s"""${potentialInput.map(_.executionName)}""", "sourceSystem" -> s"""$systemName""")(s"Writing item to flow control queue table"))
       dependencies.dynamoClient
         .writeItem(
           DADynamoDbWriteItemRequest(
```

Comment:
This wasn't part of the original plan but can you add `executionName` to the Map that gets passed to `writeItem` so if anything goes wrong we know which step function wrote to the table.